main/src/main/scala/sbt/Defaults.scala

```diff
@@ -566,11 +566,18 @@ object Defaults extends BuildCommon {
 
   def selectedFilter(args: Seq[String]): Seq[String => Boolean] =
     {
-      val filters = args map GlobFilter.apply
-      if (filters.isEmpty)
+      val (excludeArgs, includeArgs) = args.partition(s => s.startsWith("-"))
+
+      val includeFilters = includeArgs map GlobFilter.apply
+      val excludeFilters = excludeArgs.map(_.substring(1)).map(GlobFilter.apply)
+
+      if (includeFilters.isEmpty && excludeArgs.isEmpty)
         Seq(const(true))
-      else
-        filters.map { f => (s: String) => f accept s }
+      else if (includeFilters.isEmpty) {
+        excludeFilters.map { f => (s: String) => !f.accept(s) }
```

Comment:
I'm not sure this works the way you want it to.  Are all these filters "summed" up at the end, or "productized"?

i.e. you may want a single function that takes the string, and then runs through all the excludes, like:

```scala
Seq( { (s: String) => excludeFilters forall (f => !(f accept s)) }
```